### PR TITLE
(not so) Massive Issue Clearance

### LINKED ISSUE #127: Purge Command
`s!purge [amount] [@user]` - Deletes X messages (default: 50)

Requires `MANAGE_MESSAGES`.

### LINKED ISSUE #277: Add # of guilds connected to as a presence state to the bot

### LINKED ISSUE #709: Make triggered command apply the effect on the uploaded image
**Is your feature request related to a problem? Please describe.**
Yes and No, right now if you use the command triggered i'll apply the effect to your avatar, it doesn't even check if you've sent a image with the command.

**Describe the solution you'd like**
Case a image as sent with the command it should apply the effect to the image and not the user avatar.

**Additional context**
It would be nice to it also detect links like NotSoBot.

### LINKED ISSUE #714: Pirate Speak Translation Command
`s!piratespeak <text>` - Translates english text to pirate speak

**Useful packages**
https://www.npmjs.com/package/pirate-speak
https://www.npmjs.com/package/talk-like-a-pirate

### PR DESCRIPTION
**Pull request by dpaiv0**
- Closes #714
- Closes #277
- Closes #127
- Closes #709

### CODE REVIEW

File: src/commands/memes/piratespeak.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const pirateSpeak = require('pirate-speak')
+
+module.exports = class PirateSpeak extends Command {
+  constructor (client) {
+    super(client, {
+      name: 'piratespeak',
+      aliases: ['ps', 'yarrspeak'],
+      category: 'memes',
+      parameters: [{
+        type: 'string', full: true, missingError: 'commands:piratespeak.missingSentence'
```
Comment (metehus) [inline src/commands/memes/piratespeak.js]: dont forget to use the clean content

File: src/commands/moderation/purge.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      aliases: ['prune'],
+      category: 'moderation',
+      requirements: { guildOnly: true, botPermissions: ['MANAGE_MESSAGES'], permissions: ['MANAGE_MESSAGES'] },
+      parameters: [{
+        type: 'number', required: false, full: false
+      }]
+    })
+  }
+
+  async run ({ channel, guild, author, t }, number) {
```
Comment (davipatury) [inline src/commands/moderation/purge.js]:
```suggestion
  async run ({ channel, guild, author, t }, number = 50) {
```

File: src/commands/moderation/purge.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    channel.startTyping()
+    const embed = new SwitchbladeEmbed(author)
+    channel.bulkDelete(number || 50).then(() => {
```
Comment (davipatury) [inline src/commands/moderation/purge.js]:
```suggestion
    channel.bulkDelete(number).then(() => {
```

File: src/listeners/MainListener.js [hunk 4]
```diff
@@ -1,16 +1,39 @@
...
+        name: `${MiscUtils.formatNumber(this.users.size, 'en-US')} Users`,
+        type: 'WATCHING'
+      }
+    ]
+
+    setInterval(function () {
+      const presence = presences[Math.floor(Math.random() * presences.length)]
+      client.user.setPresence({ game: presence })
+      client.log(`[32mChanged presence to "${presence.name}", type "${presence.type}"`, 'Presence')
+    }, PRESENCE_INTERVAL)
```
Comment (davipatury) [inline src/listeners/MainListener.js]: NO! USE AN ARROW FUNCTION SO YOU DON'T HAVE TO DEFINE A `client` VARIABLE. THANK YOU VERY MUCH!

Comment (perronosaurio) [reply src/listeners/MainListener.js]: Ｒ　ｅ　Ｌ　ａ　ｘ

**Comment by pedrofracassi** [comment]
Stop making PRs for multiple things!!!!!

File: src/listeners/MainListener.js (hunk 4 above)
Comment (dpaiv0) [reply src/listeners/MainListener.js]: sure

File: src/listeners/MainListener.js [hunk 5]
```diff
@@ -11,6 +13,25 @@ module.exports = class MainListener extends EventListener {
   onReady () {
     this.user.setPresence({ game: { name: `@${this.user.username} help` } })
 
+    const presences = [
```
Comment (perronosaurio) [inline src/listeners/MainListener.js]: maybe the prefix should be in all, it's more intuitive

Comment (dpaiv0) [reply src/listeners/MainListener.js]: can be done, will do after getting my new phone

**Comment by dpaiv0** [comment]
### RFR

File: src/commands/image-manipulation/triggered.js [hunk 6]
```diff
@@ -10,14 +10,15 @@ module.exports = class Triggered extends Command {
...
       parameters: [{
-        type: 'user', full: true, required: false, acceptBot: true
+        type: 'image',
+        missingError: 'commands:morejpeg.missingImage'
       }]
     })
   }
 
-  async run ({ t, author, channel }, user = author) {
+  async run ({ t, author, channel }, image) {
```
Comment (Doges) [inline src/commands/image-manipulation/triggered.js]: where is the user parameter :que:

Comment (dpaiv0) [reply src/commands/image-manipulation/triggered.js]: no need to, since it uses ImageParameter

Comment (Doges) [reply src/commands/image-manipulation/triggered.js]: the imageparameter doesnt gets the author automatclly

Comment (Doges) [reply src/commands/image-manipulation/triggered.js]: wait, lemme check

Comment (Doges) [reply src/commands/image-manipulation/triggered.js]: nvm

**Pull request merged**